pubsub.go

```diff
@@ -454,6 +475,12 @@ func (p *PubSub) handleAddSubscription(req *addSubReq) {
 	// announce we want this topic
 	if len(subs) == 0 {
 		p.announce(sub.topic, true)
+		if p.rtConfig != nil && sub.topicProtocol != "" {
+			if err := p.rtConfig.JoinWithProtocol(sub.topic, sub.topicProtocol); err != nil {
```

Comment:
Wasn't really sure what to do here since I didn't want to throw a panic, but there's no way to return an error. The checks in Subscribe help make sure this doesn't happen though.

Reply:
this looks very redundant.

Reply:
The check looks redundant, or the whole `JoinWithProtocol` thing looks redundant?

Both appear redundant, however `JoinWithProtocol` is necessary if we don't want to break the interface for PubSubRouter and still support topic/Subscription options.

The router configuration check is necessary since other routers (e.g. FloodSub) may not have a router configuration. The subtopic check is necessary since I think it's reasonable that if your configurable router only has one option (e.g. standard gossipsub created by `NewGossipSub`) that you shouldn't need to subscribe with a topic protocol since it is implied. This has the added benefit of not breaking anything for people who have been using PubSub via NewGossipSub, NewFloodSub,.....

Reply:
We don't care if we break the router interface.
There are no users outside the library, and it's considered unstable anyway.

Reply:
sounds good. Take a look at my comment below (on withFloodSubPeers) which should help me figure out how to rewrite the interfaces for the next iteration.